src/rules/indentRule.ts

```diff
@@ -121,6 +128,161 @@ function walk(ctx: Lint.WalkContext<Options>): void {
     }
 }
 
+function checkIndentSize(params: {
+    ctx: Lint.WalkContext<Options>;
+    tabs: boolean;
+    size: number;
+    pos: number;
+    indentEnd: number;
+    line: string;
+    sourceFile: ts.SourceFile;
+}): boolean {
+    const ctx: Lint.WalkContext<Options> = params.ctx;
+    const tabs: boolean = params.tabs;
+    const size: number = params.size;
+    const pos: number = params.pos;
+    const indentEnd: number = params.indentEnd;
+    const line: string = params.line;
+    const sourceFile: ts.SourceFile = params.sourceFile;
+
+    const currentNodePos: number = pos + indentEnd;
+    let currentNode: ts.Node | undefined = recursiveGetNodeAt(sourceFile, currentNodePos);
+    let depth = 0;
+    if (currentNode !== undefined && inTsx(sourceFile)) {
+        return false;
+    }
+    if (currentNode !== undefined && currentNode.getStart() === currentNodePos) {
+        depth = getNodeDeepth(currentNode, line);
+    } else {
+        currentNode = recursiveGetNodeAtEnd(sourceFile, currentNodePos);
+        if (currentNode !== undefined) {
+            depth = getNodeDeepth(currentNode, line);
+        } else {
+            return false;
+        }
+    }
+    const expectedIndentation: string = tabs ? "\t".repeat(depth)
+            : " ".repeat(size * depth);
+    const actualIndentation = line.slice(0, indentEnd);
+    const failure = Rule.FAILURE_STRING(tabs ? "tab" : size === undefined ? "space" : `${depth * size} space`);
+    if (inStringTemplate(currentNode)) {
+        return false;
+    }
+    if (line.trim().charAt(0) === "*") {
+        return false;
+    }
+    if (expectedIndentation !== actualIndentation) {
+        ctx.addFailureAt(
+            pos,
+            indentEnd,
+            failure,
+            createIndentSizeFix(
+                pos,
+                tabs,
+                size,
+                depth,
+                indentEnd,
+            ),
+        );
+        return true;
+    }
+    return false;
+}
+
+function recursiveGetNodeAt(root: ts.Node, pos: number): ts.Node | undefined {
+    let result: ts.Node | undefined = root.getChildAt(pos);
+    if (result === undefined) {
+        ts.forEachChild(root, function cb(node: ts.Node): void {
+            if (node.getFullStart() < pos && node.getEnd() >= pos) {
+                result = recursiveGetNodeAt(node, pos);
+            }
+            ts.forEachChild(node, cb);
```

Comment:
you won't find the node if you recurse here, because the parent is already out of the range

Reply:
ok, code removed